controllers/storagecluster/storagecluster_controller.go

```diff
@@ -144,14 +150,61 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
-	builder := ctrl.NewControllerManagedBy(mgr).
+	noobaaIgnoreTimeUpdatePredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if e.ObjectOld == nil || e.ObjectNew == nil {
+				return false
+			}
+			oldObj := e.ObjectOld.(*nbv1.NooBaa)
+			newObj := e.ObjectNew.(*nbv1.NooBaa)
+
+			diff := cmp.Diff(
+				oldObj, newObj,
+				cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion"),
+				cmpopts.IgnoreFields(conditionsv1.Condition{}, "LastHeartbeatTime", "LastTransitionTime"),
+			)
+
+			return diff != ""
+		},
+	}
+
+	cephClusterIgnoreTimeUpdatePredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if e.ObjectOld == nil || e.ObjectNew == nil {
+				return false
+			}
+			oldObj := e.ObjectOld.(*cephv1.CephCluster)
+			newObj := e.ObjectNew.(*cephv1.CephCluster)
+
+			diff := cmp.Diff(
+				oldObj, newObj,
+				cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion"),
+				cmpopts.IgnoreFields(conditionsv1.Condition{}, "LastHeartbeatTime", "LastTransitionTime"),
+			)
+
+			return diff != ""
+		},
+	}
+
+	build := ctrl.NewControllerManagedBy(mgr).
```

Comment:
what's the intention of new `Owns` in this PR? is `storagecluster` already a owner of these resources?

Reply:
The storagecluster is already a owner of these resources.

Previously, any changes to the NooBaa and Ceph clusters would trigger the reconciliation process. This is no longer happening. Before, even if other custom resources (CRs) were changed, the reconciliation would still run because of the NooBaa and Ceph clusters. Now, we need to ensure that changes to other CRs still trigger the reconciliation process.

Reply:
yes, can be resolved.